lib/streamlit/runtime/context.py

```diff
@@ -307,3 +307,17 @@ def url(self) -> str | None:
         if ctx is None or ctx.context_info is None:
             return None
         return ctx.context_info.url
+
+    @property
+    @gather_metrics("context.ip_address")
+    def ip_address(self) -> str | None:
+        """The read-only IP address of the user's connection.
+        This should not be used for security measures as it can be easily spoofed.
+        """
+        session_client_request = _get_request()
+        if session_client_request is not None:
+            remote_ip = session_client_request.remote_ip
+            if remote_ip == "::1" or remote_ip == "127.0.0.1":
+                return None
+            return remote_ip
+        return None
```

Comment:
this last `return None` could be removed, but GitHub advanced security bot doesn't likes when we mix explicit and implicit returns :)